### PR TITLE
allas checks and updates

### PR DESCRIPTION
**Pull request by kkmattil**
## Proposed changes

Updates to Allas documentation. 

## Checklist before requesting a review

- [ ] I have followed the instructions in the [Contributing](https://github.com/CSCfi/csc-user-guide/blob/master/CONTRIBUTING.md) and [Styleguide](https://github.com/CSCfi/csc-user-guide/blob/master/STYLEGUIDE.md) documents.
- [ ] My pull request passes the automated tests. If not, visit [pull requests at Travis CI](https://app.travis-ci.com/github/CSCfi/csc-user-guide/pull_requests) and have a look at the job log.
- [ ] I have included a link to the appropriate [preview page](https://csc-guide-preview.2.rahtiapp.fi/origin/) (select your branch from the list).

### CODE REVIEW

File: docs/data/Allas/accessing_allas.md [hunk 1]
```diff
@@ -12,16 +12,34 @@ Once you have Allas enabled you can access it from any machine or server that is
...
 ## Accessing Allas from the web browser
 
-At the moment CSC provides two web browser interfaces for Allas:
+At the moment CSC provides several web browser interfaces for Allas:
+
+The **WWW interfaces of Puhti and Mahti** are connected to Allas. 
+These interfaces allow you transfer files and directories between your local computer and Allas as well as
+between CSC supercomputers and Allas.
+
+* [Instuctions for using Allas in Puhti and Mahti web interfaces](../../computing/webinterface/file-browser.md)
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:21]:
```suggestion
* [Instructions for using Allas in Puhti and Mahti web interfaces](../../computing/webinterface/file-browser.md)
```

File: docs/data/Allas/accessing_allas.md [hunk 2]
```diff
@@ -12,16 +12,34 @@ Once you have Allas enabled you can access it from any machine or server that is
...
+
 
 The OpenStack Horizon web interface in **cPouta** provides easy-to-use basic functions for data management in Allas. This interface can only be used for files smaller than 5 GB.
 
 * [Web client – OpenStack Horizon Dashboard](./using_allas/web_client.md)
+* [cPouta Web Interface](https://pouta.csc.fi)
+
+
+**SD Connect** provides an interface for storing and sharing sensitive data. 
+This service is based of Allas but we don't recommend it for other than sensitive data.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:33]:
```suggestion
This service is based on Allas but we don't recommend it for other than sensitive data.
```

File: docs/data/Allas/accessing_allas.md [hunk 3]
```diff
@@ -12,16 +12,34 @@ Once you have Allas enabled you can access it from any machine or server that is
...
+
+* [SD Connect instructions](../sensitive-data/sd_connect.md)
+* [SD Connect interface](https://sd-connect.csc.fi)
+
 
 
 ## Accessing Allas in the CSC computing environment and other Linux platforms
 
-Puhti and Mahti servers at CSC support many different tools for using Allas, These include
+Puhti and Mahti servers at CSC support many different command line tools for using Allas, These include
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:42]:
```suggestion
CSC supercomputers Puhti and Mahti support many different command line tools for using Allas, These include
```

File: docs/data/Allas/accessing_allas.md [hunk 4]
```diff
@@ -30,38 +48,39 @@ Puhti and Mahti servers at CSC support many different tools for using Allas, The
 
 Note that the tools listed above utilize two different protocols: _Swift_ and _S3_. Data uploaded using one protocol is not necessary compatible with another protocol. 
 
-The software listed above can be used in other Linux servers as well, e.g. a virtual machine running in cPouta or your own Linux-based laptop. In that case, you need to install the client software and configure the connection to Allas yourself. Instructions : [allas-cli-utils](https://github.com/CSCfi/allas-cli-utils)
+The software listed above can be used other machines too like in a virtual machine running in cPouta or your own laptop.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:51]:
```suggestion
The software listed above can also be used on other devices, for example a virtual machine running in cPouta or your own laptop.
```

File: docs/data/Allas/accessing_allas.md [hunk 5]
```diff
@@ -30,38 +48,39 @@ Puhti and Mahti servers at CSC support many different tools for using Allas, The
...
 ```
-The _allas-conf_ command prompts for your CSC password (the same that you use to login to CSC servers). It lists your Allas projects and asks you to define a project (if not already defined as an argument). _allas-conf_ generates an _rclone_ configuration file for the Allas service and authenticates the connection to the selected project. You can only be connected to one Allas project at a time in one session. The project you are using in Allas does not need to match the project you are using in Puhti or Mahti, and you can switch to another project by running _allas-conf_ again.
+The _allas-conf_ command prompts for your CSC password (the same that you use to login to CSC servers). It lists your Allas projects and asks you to define a project (if not already defined as an argument). _allas-conf_ generates an _rclone_ configuration file for the Allas service and authenticates the connection to the selected project. allas-conf enables you to use only one Allas project at a time in one session. The project you are using in Allas does not need to match the project you are using in Puhti or Mahti, and you can switch to another project by running _allas-conf_ again.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:66]:
```suggestion
The `allas-conf` command prompts for your CSC password (the same that you use to login to CSC servers). It lists your Allas projects and asks you to define a project (if not already defined as an argument). `allas-conf` generates an `rclone` configuration file for the Allas service and authenticates the connection to the selected project. `allas-conf` enables you to use only one Allas project at a time in one session. The project you are using in Allas does not need to match the project you are using in Puhti or Mahti, and you can switch to another project by running `allas-conf` again.
```

File: docs/data/Allas/accessing_allas.md [hunk 6]
```diff
@@ -30,38 +48,39 @@ Puhti and Mahti servers at CSC support many different tools for using Allas, The
...
 
-Authentication information is stored in the shell variables *OS_AUTH_TOKEN* and *OS_STORAGE_URL* and is valid for up to eight hours. However, you can refresh the authentication at any time my running _allas-conf_ again. The environment variables are available only for that login session, so if you start another shell session, you need to authenticate again in there to access Allas.
+In the case of swift-protocol, the authentication information is stored in the shell variables *OS_AUTH_TOKEN* and *OS_STORAGE_URL* and is valid for up to eight hours. However, you can refresh the authentication at any time my running _allas-conf_ again. The environment variables are available only for that login session, so if you start another shell session, you need to authenticate again in there to access Allas.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:68]:
```suggestion
In the case of the Swift protocol, the authentication information is stored in the `OS_AUTH_TOKEN` and `OS_STORAGE_URL` environment variables and is valid for up to eight hours. However, you can refresh the authentication at any time by running `allas-conf` again. The environment variables are set only for the current login session, so you need to configure authentication individually for each shell with which you wish to access Allas.
```

File: docs/data/Allas/accessing_allas.md [hunk 7]
```diff
@@ -30,38 +48,39 @@ Puhti and Mahti servers at CSC support many different tools for using Allas, The
...
 
-Once Allas connection is congigured, you can start using Allas with the tools listed above. 
+In the case of S3 protocol, the authentication information is stored to configuration files in your home directory. Thus the authentication will be used for all sessions in in the machine in uses. The authentication does not have expiration time.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:70]:
```suggestion
In the case of the S3 protocol, the authentication information is stored in configuration files located in your home directory on the device. The same authentication is used for all login sessions and it does not have an expiration time.
```

File: docs/data/Allas/accessing_allas.md [hunk 8]
```diff
@@ -30,38 +48,39 @@ Puhti and Mahti servers at CSC support many different tools for using Allas, The
...
+
+Once Allas connection is configured, you can start using Allas with the tools listed above.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:72]:
```suggestion
Once an Allas connection is configured, you can start using the object storage with the tools listed above. 
```

File: docs/data/Allas/accessing_allas.md [hunk 9]
```diff
@@ -70,8 +89,8 @@ In addition to the Web interfaces listed above, you can access Allas from you Wi
 For example following tools can be used:
 
 * [Cyberduck](./using_allas/cyberduck.md) provides easy to use graphical interface for moving data between local computer and Allas.
-* [Rclone](./using_allas/rclone_local.md) is an command line tool that provides very effective way to use Allas.
-* [a-tools](./using_allas/a_commands.md) these Allas specific commands can be installed in Mac OSX machines but not to Windows
+* [Rclone](./using_allas/rclone_local.md) is an command line tool that provides very effective way to use Allas from any operating system.
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:92]:
```suggestion
* [Rclone](./using_allas/rclone_local.md) is a command line tool that provides a very effective way to use Allas on any operating system.
```

File: docs/data/Allas/accessing_allas.md [hunk 10]
```diff
@@ -70,8 +89,8 @@ In addition to the Web interfaces listed above, you can access Allas from you Wi
...
+* [a-tools](./using_allas/a_commands.md) these Allas specific commands can be installed in Mac OSX and Linux machines but not to Windows
```
Comment (msainio) [inline docs/data/Allas/accessing_allas.md:93]:
```suggestion
* [a-tools](./using_allas/a_commands.md) are Allas-specific commands that can be installed on macOS and Linux devices, but not ones running a Windows operating system.
```

File: docs/data/Allas/allas_encryption.md [hunk 11]
```diff
@@ -1,11 +1,14 @@
 # Tools for client side encryption for Allas
 
 Allas is not certified as high level security storage platform and thus you should not use it to store sensitive data in readable format.
-Howerver, sensitive data can be stored to Allas if it is properly encrypted before data is transported to Allas.
+Howerver, sensitive data can be stored to Allas if it is properly encrypted before data is transported to Allas.
```
Comment (msainio) [inline docs/data/Allas/allas_encryption.md:4]:
```suggestion
However, sensitive data can be stored on Allas if it is properly encrypted before transfer to the object storage.
```

File: docs/data/Allas/allas_encryption.md [hunk 12]
```diff
@@ -1,11 +1,14 @@
 # Tools for client side encryption for Allas
 
 Allas is not certified as high level security storage platform and thus you should not use it to store sensitive data in readable format.
-Howerver, sensitive data can be stored to Allas if it is properly encrypted before data is transported to Allas.
+Howerver, sensitive data can be stored to Allas if it is properly encrypted before data is transported to Allas. 
 
-This document describes some encryption tools that help you to move your sensitive data to Allas so that data gets encrypted before it leaves your secure environment. When you use Allas with these encryption tools, remember that:
+For sensitive data we generally recommend the [SD Connect service](../sensitive-data/sd_connect.md) that provides Web and command line interfaces 
+that automatically encrypt the data when it stored to Allas.
```
Comment (msainio) [inline docs/data/Allas/allas_encryption.md:7]:
```suggestion
that automatically encrypt the data when it is stored to Allas. 
```

File: docs/data/Allas/allas_encryption.md [hunk 13]
```diff
@@ -1,11 +1,14 @@
...
 
 Allas is not certified as high level security storage platform and thus you should not use it to store sensitive data in readable format.
-Howerver, sensitive data can be stored to Allas if it is properly encrypted before data is transported to Allas.
+Howerver, sensitive data can be stored to Allas if it is properly encrypted before data is transported to Allas. 
 
-This document describes some encryption tools that help you to move your sensitive data to Allas so that data gets encrypted before it leaves your secure environment. When you use Allas with these encryption tools, remember that:
+For sensitive data we generally recommend the [SD Connect service](../sensitive-data/sd_connect.md) that provides Web and command line interfaces 
+that automatically encrypt the data when it stored to Allas. 
+
+If you for some reason don't want to use SD Connect below you can find some alternative approaches for storing your sensitive data to Allas. When you use Allas with these encryption tools, remember that:
```
Comment (msainio) [inline docs/data/Allas/allas_encryption.md:9]:
```suggestion
If you for some reason don't want to use SD Connect, below you can find some alternative approaches for storing your sensitive data to Allas. When you use Allas with these encryption tools, remember that:
```

File: docs/data/Allas/allas_encryption.md [hunk 14]
```diff
@@ -1,11 +1,14 @@
...
    
-   1. You can store encrypted sensitive data in Allas, but are allowed to decrypt it only in evironments with high enough security level. For example, in the HPC environment of CSC is NOT secure enough for sensitive data. 
+   1. You can store encrypted sensitive data in Allas, but are allowed to decrypt it only in environments with high enough security level. For example, in the HPC environment of CSC is NOT secure enough for sensitive data.
```
Comment (msainio) [inline docs/data/Allas/allas_encryption.md:11]:
```suggestion
   1. You can store encrypted sensitive data in Allas, but are allowed to decrypt it only in environments with a high enough security level. For example, the CSC HPC environment (i.e. Puhti, Mahti, LUMI) is **not** secure enough for sensitive data.
```

File: docs/data/Allas/allas_encryption.md [hunk 15]
```diff
@@ -37,10 +40,9 @@ a-get my_allas_bucket/data_dir.tar.zst.gpg
 ### Asymmetric crypt4gh encryption
 
 If you want to use asymmetric `crypt4gh` encryption, you need to have a public key file for encryption and a secret key file for decryption.
-In Puhti, you first need to make `crypt4gh` available with commands:
+In Puhti, you first need to make `crypt4gh` available with command:
```
Comment (msainio) [inline docs/data/Allas/allas_encryption.md:43]:
```suggestion
In Puhti, you first need to make `crypt4gh` available with the command:
```

File: docs/data/Allas/allas_project_example.md [hunk 16]
```diff
@@ -85,17 +85,17 @@ Because storing data in Allas consumes billing units, Saara needs to check the s
 
 Allas storage is only for research project's duration, but Saara thinks it would be beneficial to have the preliminary data made publicly available and easier to be found. This is supported by the [Fairdata Services](https://www.fairdata.fi/en/) produced by CSC.
 
-Pekka creates a new bucket with public access and uploads the data to the bucket. Command _a-publish_ creates such new bucket and uploads a files in to it. Parameter -b is used to define the name for the bukcet, in this case _hiano-project-public001_.
+Pekka creates a new bucket with public access and uploads the data to the bucket. Command _a-publish_ creates such new bucket and uploads a files in to it. Parameter -b is used to define the name for the bucket, in this case _hiano-project-public001_.
```
Comment (msainio) [inline docs/data/Allas/allas_project_example.md:88]:
```suggestion
Pekka creates a new bucket with public access and uploads the data to the bucket. Command _a-publish_ creates the bucket and uploads the selected files to it. Parameter `-b` is used to define the name for the bucket, in this case `hiano-project-public001`.
```

File: docs/data/Allas/allas_project_example.md [hunk 17]
```diff
@@ -85,17 +85,17 @@ Because storing data in Allas consumes billing units, Saara needs to check the s
...
 ```
-Next Pekka describes basic information about the data in [Fairdata Qvain Tool](https://www.fairdata.fi/en/qvain/) and provides the two URLs (one for each file in Allas) as a Remote Resource in Qvain. After this the data can then be published as a dataset with a landing page and a persistent identifier. This way the preliminary data can be shared among colleaques using this persistent identifier. The dataset can also be explored via [Fairdata Etsin Service](https://www.fairdata.fi/en/etsin/) with structured information and direct access to download the files in Allas.
+Next Pekka describes basic information about the data in [Fairdata Qvain Tool](https://www.fairdata.fi/en/qvain/) and provides the two URLs (one for each file in Allas) as a Remote Resource in Qvain. After this the data can then be published as a dataset with a landing page and a persistent identifier. This way the preliminary data can be shared among colleagues using this persistent identifier. The dataset can also be explored via [Fairdata Etsin Service](https://www.fairdata.fi/en/etsin/) with structured information and direct access to download the files in Allas.
```
Comment (msainio) [inline docs/data/Allas/allas_project_example.md:92]:
```suggestion
Next, Pekka creates a basic description of the data using the [Fairdata Qvain Tool](https://www.fairdata.fi/en/qvain/) and provides the two URLs (one for each file in Allas) as a Remote Resource in Qvain. After this, the data can be published as a dataset with a landing page and a persistent identifier. This way the preliminary data can be shared among colleagues using the persistent identifier. The dataset can also be explored via [Fairdata Etsin Service](https://www.fairdata.fi/en/etsin/) with structured information and direct access to download the files in Allas.
```

File: docs/data/Allas/using_allas/rclone.md [hunk 18]
```diff
@@ -1,6 +1,6 @@
 # Using Allas with Rclone 
 
-This chapter contains instructions for using Allas with [Rclone](https://rclone.org/) in the Puhti and Mahti computing environments. _Rclone_ provides a very powerful and versatile way to use Allas and other object storage services. It is able to use both the S3 and Swift protocols (and many others), but in the case of Allas, the Swift protocol is preferred. It is also the default option on the CSC servers. 
+This chapter contains instructions for using Allas with [Rclone](https://rclone.org/) in the Puhti and Mahti computing environments. _Rclone_ provides a very powerful and versatile way to use Allas and other object storage services. It is able to use both the S3 and Swift protocols (and many others). At the moment Swift protocol is the default option on the CSC servers.
```
Comment (msainio) [inline docs/data/Allas/using_allas/rclone.md:3]:
```suggestion
This chapter contains instructions for using Allas with [Rclone](https://rclone.org/) in the Puhti and Mahti computing environments. _Rclone_ provides a very powerful and versatile way to use Allas and other object storage services. It is able to use both the S3 and Swift protocols (and many others). At the moment, the Swift protocol is the default option on the CSC servers. 
```

File: docs/data/Allas/using_allas/rclone.md [hunk 19]
```diff
@@ -49,23 +49,23 @@ The default protocol of Allas is Swift. In Puhti and Mahti Swift based Allas con
 module load allas
 allas-conf
 ```
-The `allas-conf` command asks for your CSC password (the same you use to login to CSC servers). It lists
+The `allas-conf` command asks for your CSC password (University/Haka password will not work here). It lists
 your projects in Allas and asks you to define the project that will be used. Then _allas-conf_ generates a Rclone configuration file for the Allas service and authenticates the connection to the selected project. In Rclone command this swift based connection is referred with remote name `allas:`. The authentication information is stored in the shell variables `OS_AUTH_TOKEN` and `OS_STORAGE_URL` that are valid for up to eight hours. However, you can refresh the authentication at any time by running _allas-conf_ again. The environment variables are available only for that login session, so if you login to Puhti in another session, you need to authenticate again to access Allas.
 
 ### Rclone with S3 on CSC supercomputers
 
 If you want to use Allas with S3 protocol in stead run the allas-conf command with option --mode s3cmd.
```
Comment (msainio) [inline docs/data/Allas/using_allas/rclone.md:57]:
```suggestion
If you want to use Allas with the S3 protocol instead, run the allas-conf command with the `--mode S3` option.
```

Comment (msainio) [reply docs/data/Allas/using_allas/rclone.md:57]: Since the example below was changed from `s3cmd` to `S3`.

Comment (kkmattil) [reply docs/data/Allas/using_allas/rclone.md]: Good

File: docs/data/Allas/using_allas/rclone.md [hunk 20]
```diff
@@ -81,7 +81,7 @@ is deleted after copying.
...
+During upload files that are larger than 5 GB will be split and stored as several objects. This is done automatically to a bucket that is named by adding extension `_segments` to the original bucket name. For example, if you would upload a large file to  _2000620-raw-data_ the actual data would be stored as several pieces into bucket _2000620-raw-data_segments_. The target bucket (_2000620-raw-data_) would contain just a manifest object that contains information what segments make the stored file. Operations performed to the manifest object are automatically reflected to the segments. Normally users don't need to operate with the segments buckets at all and objects inside these buckets should not be deleted or modified.
```
Comment (msainio) [inline docs/data/Allas/using_allas/rclone.md:84]:
```suggestion
During upload, files that are larger than 5 GB will be split and stored as several objects. The objects are stored automatically in a distinct bucket called `<bucket-name>_segments`. For example, if you would upload a large file to  `2000620-raw-data`, the actual data would be stored in several pieces in the bucket `2000620-raw-data_segments`. The target bucket (`2000620-raw-data`) would contain just a manifest object stating which segments comprise the stored file. Operations performed on the manifest object are automatically reflected in the segments. Normally users don't need to operate with the segments buckets at all, and objects inside these buckets should not be deleted or modified.
```

**Comment by msainio** [comment]
Is `--mode S3` now preferred over `--mode s3cmd`?

**Comment by kkmattil** [comment]
> Is `--mode S3` now preferred over `--mode s3cmd`?

s3cmd, s3 and S3 all do the same thing. I think `--mode S3` is most intuitive.

**Review by msainio — APPROVED** [review]
The amount of documented approaches is extended, and existing instructions are brought up to date. Good changes!

**Pull request merged**